src/test/java/jenkins/plugins/http_request/Registers.java

```diff
@@ -278,37 +294,44 @@ static void registerFileUpload(final File testFolder, final File uploadFile, fin
 
 			private static final String MULTIPART_FORMDATA_TYPE = "multipart/form-data";
 
-			private void enableMultipartSupport(HttpServletRequest request, MultipartConfigElement multipartConfig) {
-				request.setAttribute(Request.__MULTIPART_CONFIG_ELEMENT, multipartConfig);
+			private void enableMultipartSupport(Request request, MultipartConfigElement multipartConfig) {
+				request.setAttribute("org.eclipse.jetty.multipartConfig", multipartConfig);
 			}
 
-			private boolean isMultipartRequest(ServletRequest request) {
-				return request.getContentType() != null && request.getContentType().startsWith(MULTIPART_FORMDATA_TYPE);
+			private boolean isMultipartRequest(Request request) {
+				return request.getHeaders().get(HttpHeaders.CONTENT_TYPE) != null && request.getHeaders().get(HttpHeaders.CONTENT_TYPE).startsWith(MULTIPART_FORMDATA_TYPE);
 			}
 
 			@Override
-			void doHandle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+			boolean doHandle(Request request, Response response, Callback callback) throws IOException {
+				String MULTIPART = "org.eclipse.jetty.servlet.MultiPartFile.multiPartInputStream";
+				MultiPartFormInputStream multipartInputStream = (MultiPartFormInputStream) request.getAttribute(MULTIPART);
+
 				assertEquals("POST", request.getMethod());
 				assertTrue(isMultipartRequest(request));
 
 				MultipartConfigElement multipartConfig = new MultipartConfigElement(testFolder.getAbsolutePath());
 				enableMultipartSupport(request, multipartConfig);
 
 				try {
-					Part part = request.getPart("file-name");
-					assertNotNull(part);
-					assertEquals(uploadFile.length(), part.getSize());
-					assertEquals(uploadFile.getName(), part.getSubmittedFileName());
-					assertEquals(MimeType.APPLICATION_ZIP.getValue(), part.getContentType());
+					if (multipartInputStream != null) {
```

Comment:
Try setting a breakpoint here and running the test suite. In my tests `multipartInputStream` is always null, and the code inside the `if` statement is never executed. As such, this turns the test into a false negative—the assertions are never executed. Presumably the assertions were executed before, making this a regression.

Reply:
Thank you for the catch, yes `multipartInputStream` is always null on my side too.
With what @olamy suggested here https://github.com/BorisYaoA/http-request-plugin/pull/1, I do not use it anymore and it is more clear.

And Just FTR I've run the same test in debug mode on the actual `master` branch and multipartInputStream is always null so it seems that the `finally` here https://github.com/jenkinsci/http-request-plugin/blob/018c44c33198600160f08be7cb4a0bc7ded19182/src/test/java/jenkins/plugins/http_request/Registers.java#L364-L371 wasn't doing nothing